TechnicoBackend/Controllers/AuthController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using TechnicoBackend.Repositories;
+using TechnicoBackend.Models;
+using TechnicoBackend.DTOs;
+
+namespace TechnicoBackend.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class AuthController : ControllerBase
+    {
+        private readonly UserRepository _userRepository;
+
+        public AuthController(UserRepository userRepository)
+        {
+            _userRepository = userRepository;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(UserDTO userDto)
+        {
+            try
+            {
+                var users = await _userRepository.GetAllAsync();
+                if (users.Any(u => u.Email == userDto.Email))
+                {
+                    return BadRequest("Το email χρησιμοποιείται ήδη.");
+                }
+
+                var user = new User
+                {
+                    Email = userDto.Email,
+                    PhoneNumber = userDto.PhoneNumber,
+                    FirstName = userDto.FirstName ?? string.Empty,
+                    LastName = userDto.LastName ?? string.Empty,
+                    Password = userDto.Password ?? string.Empty,
+                    VatNumber = userDto.VatNumber ?? string.Empty,
+                    UserType = "PropertyOwner"
+                };
+
+                await _userRepository.AddAsync(user);
+
+                return Ok(new
+                {
+                    Id = user.Id,
+                    Message = "Η εγγραφή ολοκληρώθηκε με επιτυχία."
+                });
+            }
+            catch (Exception ex)
+            {
+                return BadRequest($"Σφάλμα κατά την εγγραφή: {ex.Message}");
+            }
+        }
+
+        [HttpPost("login")]
+        public async Task<IActionResult> Login(LoginDTO loginDto)
```

Comment:
Consider using a Service to implement any business logic/functionality. Consider using Controller to only receive the data and call the service.